aspnetcore/blazor/components/integration.md

```diff
@@ -22,24 +22,171 @@ Razor components can be integrated into Razor Pages, MVC, and other types of ASP
+    <base href="/" />
+    <link rel="stylesheet" href="{APP NAMESPACE}.styles.css" />
+    <link rel="icon" type="image/png" href="favicon.png" />
+    <HeadOutlet />
+</head>
+
+<body>
+    <Routes />
+    <script src="_framework/blazor.web.js"></script>
+</body>
+
+</html>
+```
+
+`Components/Pages/Welcome.razor`:
+
+```razor
+@page "/welcome"
+
+<PageTitle>Welcome!</PageTitle>
+
+<h1>Welcome to Blazor!</h1>
+
+<p>@message</p>
+
+@code {
+    private string message = 
+        "Hello from a Razor component and welcome to Blazor!";
+}
+```
+
+In the ASP.NET Core project's `Program` file, add a `using` statement to the top of the file for the project's components:
+
+```csharp
+using {APP NAMESPACE}.Components;
+```
+
+Add Razor component services (`AddRazorComponents`) before the app is built (the line that calls `builder.Build()`):
+
+```csharp
+builder.Services.AddRazorComponents();
+```
+
+Add [Antiforgery Middleware](xref:blazor/security/index#antiforgery-support) to the request processing pipeline after the call to `app.UseRouting`. If there are calls to `app.UseRouting` and `app.UseEndpoints`, the call to `app.UseAntiforgery` must go between them. A call to `app.UseAntiforgery` must be placed after calls to `app.UseAuthentication` and `app.UseAuthorization`.
+
+```csharp
+app.UseAntiforgery();
+```
+
+Add `MapRazorComponents` to the app's request processing pipeline with the `App` component (`App.razor`) specified as the default root component. Place the following code before the app is run (the line that calls `app.Run`):
+
+```csharp
+app.MapRazorComponents<App>();
+```
+
+When the app is run, the `Welcome` component is accessed at the `/welcome` endpoint.
+
+### Enable interactive server rendering
+
+Follow the guidance in the [Enable interactive server rendering](#enable-interactive-server-rendering) section.
+
+Make the following changes in the app's `Program` file.
+
+Add a call to `AddInteractiveServerComponents` where Razor component services are added with `AddRazorComponents`:
+
+```csharp
+builder.Services.AddRazorComponents()
+    .AddInteractiveServerComponents();
+```
+
+Add a call to `AddInteractiveServerRenderMode` where Razor components are mapped with `MapRazorComponents`:
+
+```csharp
+app.MapRazorComponents<App>()
+    .AddInteractiveServerRenderMode();
+```
+
+Add a `Counter` component to the app with the interactive server render mode.
+
+`Components/Pages/Counter.razor`:
+
+```razor
+@page "/counter"
+@attribute [RenderModeInteractiveServer]
+
+<PageTitle>Counter</PageTitle>
+
+<h1>Counter</h1>
+
+<p role="status">Current count: @currentCount</p>
+
+<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>
+
+@code {
+    private int currentCount = 0;
+
+    private void IncrementCount()
+    {
+        currentCount++;
+    }
+}
+```
+
+When the app is run, the `Counter` component is accessed at the `/counter` endpoint.
+
+### Enable interactive Auto or WebAssembly rendering
```

Comment:
I'd also like to simplify this section to make it more minimalistic:
- Clarify at the beginning that the user should first enable support for static server rendering if they don't have that already, like we do in the section on enabling interactive server rendering.
- When creating the donor project use the `--empty` option. We're just trying to enable the project structure, not include the sample pages and content.
- Remove step 2 to copy over Bootstrap and the styles. The existing app might not be using Bootstrap.
- Remove step 4. The generated project without sample pages won't have a Components folder. We'll need to add a step to add a component to the client project to demonstrate that it's working.
- Simplify step 10 given that the user has done a bunch of the described changes already as part of enabling static server rendering. Key things are to enable interactive WebAssembly support and to add the client project as an additional assembly
- Step 11 is huge! 😮 I'd like to shrink it way down to just adding another *Pages/Counter2.razor* component to the client project with `@rendermode InteractiveWebAssembly` or `InteractiveAuto`.
-

Reply:
Ok ... I'll put it on the new PR tho. That way, all lines go in clean for feedback.